masonry_winit/src/event_loop_runner.rs

```diff
@@ -68,12 +68,19 @@ pub struct NewWindow {
 }
 
 impl NewWindow {
-    /// Create a new window with auto assigned [`WindowId::next()`].
+    /// A default `NewWindow`.
+    ///
+    /// Further properties of the window which will be created using
+    /// builder-style methods on this type.
+    /// See the documentation on the fields of this type for details of the parameters.
     pub fn new(attributes: WindowAttributes, root_widget: NewWidget<dyn Widget + 'static>) -> Self {
         Self::new_with_id(WindowId::next(), attributes, root_widget)
     }
 
-    /// Create a new window with a custom assigned [`WindowId`].
+    /// Create a `NewWindow` with a custom assigned [`WindowId`].
+    ///
+    /// Use this when you need to specify a unique ID for the window, for example,
+    /// for external tracking or state management.
```

Comment:
It isn't clear to me what use cases there are for creating a window in Masonry Winit, but not knowing its ID?

And so I think:
```rust
let id = WindowId::next();
create_window(NewWindow::new(id, ..., ...));
```
is clearer than:
```
let new_window = NewWindow::new(..., ...);
let id = new_window.id;
create_window(new_window);
```

I think this splitting of methods probably should have been combined with making `create_window` return the id of the created window.

(This shouldn't be actioned in this PR - we probably should just split this into an issue, or do it in your follow-up)